enterprise/frontend/src/metabase-enterprise/upload_management/UploadManagementTable.tsx

```diff
@@ -0,0 +1,199 @@
+import { useCallback, useState } from "react";
+import { msgid, ngettext, t } from "ttag";
+
+import SettingHeader from "metabase/admin/settings/components/SettingHeader";
+import { StyledTable } from "metabase/common/components/Table";
+import {
+  BulkActionBar,
+  BulkActionButton,
+} from "metabase/components/BulkActionBar";
+import { DelayedLoadingAndErrorWrapper } from "metabase/components/LoadingAndErrorWrapper/DelayedLoadingAndErrorWrapper";
+import Link from "metabase/core/components/Link";
+import { useDispatch } from "metabase/lib/redux";
+import * as Urls from "metabase/lib/urls";
+import { addUndo } from "metabase/redux/undo";
+import { Box, Flex, Text, Button, Icon, Checkbox } from "metabase/ui";
+import {
+  useListUploadTablesQuery,
+  useDeleteUploadTableMutation,
+} from "metabase-enterprise/api";
+import type { Table } from "metabase-types/api";
+
+import { DeleteConfirmModal } from "./DeleteConfirmModal";
+import { getDateDisplay } from "./utils";
+
+const columns = [
+  { key: "checkbox", name: "" },
+  { key: "name", name: t`Table name` },
+  { key: "created_at", name: t`Created at` },
+  { key: "schema", name: t`Schema` },
+  { key: "actions", name: "" },
+];
+
+export function UploadManagementTable() {
+  const [selectedItems, setSelectedItems] = useState<Table[]>([]);
+  const [deleteTableRequest] = useDeleteUploadTableMutation();
+  const [showDeleteConfirmModal, setShowDeleteConfirmModal] = useState(false);
+  const dispatch = useDispatch();
+
+  const { data: uploadTables, error, isLoading } = useListUploadTablesQuery();
+
+  const deleteTable = useCallback(
+    async (table: Table, sendToTrash: boolean) => {
+      return deleteTableRequest({
+        tableId: table.id,
+        "archive-cards": sendToTrash,
+      });
+    },
+    [deleteTableRequest],
+  );
+
+  const renderRow = useCallback(
+    (row: Table) => (
+      <UploadTableRow
+        item={row}
+        isSelected={selectedItems.some(
+          selectedTable => selectedTable.id === row.id,
+        )}
```

Comment:
not to be that guy, but this is n^2. would prefer adding this above 
```
const selectedItemIds = useMemo(() => {
   return new Set(selectedItems.map(item => item.id));
}, [selectedItems]);
```
then doing this here instead
```
selectItemIds.has(row.id)
```